### DISCUSSION TITLE
German Umlauts Not Detected (Ä, Ö, Ü) in PaddleOCR

### DISCUSSION THREAD

**Discussion by sayinmehmet47** [PaddlePaddle/PaddleOCR#14792] [Q&A]
Hi everyone,

I have implemented PaddleOCR to extract text from insurance cards. However, I noticed that some German umlauts (e.g., Ä, Ö, Ü) are not being correctly detected. I also updated the version of paddle ocr to latest , it also did not worked

For example, the word "EUROPÄISCHE" is recognized as "EUROPAISCHE", missing the Ä. `Dömer` detected as `Domer`

How can I improve the OCR model to correctly recognize German umlauts? Do I need to fine-tune the model, or is there a configuration I can adjust?

To help debug, I created a dedicated repository showing how I implemented OCR:

https://github.com/sayinmehmet47/ocr

I also have a testable endpoint available:

Any suggestions or guidance would be greatly appreciated!

Thanks in advance!

**Comment by GreatV** [comment]
It looks like PaddleOCR is not recognizing German umlauts (Ä, Ö, Ü) correctly in your use case. Here are a few steps you can take to improve the model's accuracy for German characters:

### **1. Check the Language Model Configuration**
PaddleOCR uses different pre-trained models for different languages. Ensure that you are using a model that supports German characters. You can try using the multilingual model (`ppocr_v3`) or explicitly specify the German character set. 

Modify the `ocr_config` to include a custom character list that includes umlauts:
```python
custom_dict = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzÄÖÜäöüß"
```
You may also need to provide this dictionary in the configuration when initializing PaddleOCR.

### **2. Use a Custom Character List**
If the default model does not properly recognize umlauts, you can modify the `dict.txt` file used during training. In PaddleOCR, changing the character dictionary can help the model recognize additional characters.

### **3. Try a Different Pre-trained Model**
Some models may have better support for different languages. You could try initializing the OCR system with a different trained model and see if recognition improves:
```python
ocr = PaddleOCR(lang="de")
```
If `lang="de"` isn't available, try using `"en"` or `"multi"` and customize the dictionary.

### **4. Fine-tune the Model**
If the pre-trained models do not correctly recognize umlauts, fine-tuning the OCR model on a dataset that includes German text with umlauts could improve performance. You would need to:
   - Collect images with German text containing umlauts.
   - Annotate them correctly.
   - Fine-tune a PaddleOCR model using those samples.

### **5. Preprocessing the Image**
If the umlauts are not detected, try enhancing the image quality:
   - Convert images to grayscale
   - Increase contrast
   - Apply adaptive thresholding
   - Experiment with different resolutions

### **6. Post-processing Correction**
You can use a dictionary-based correction approach. After OCR extraction, compare words with a dictionary of common words and correct missing umlauts accordingly.

### **7. Check Recent Issues and Discussions**
The discussed Czech language issue ([Discussion](https://github.com/PaddlePaddle/PaddleOCR/discussions/13914)) indicates similar diacritic-related problems. You may want to check if other users have identified solutions for non-English character recognition.

If none of these steps resolve the issue, a custom training approach may be necessary.

Would you be able to share some sample OCR outputs along with images to further debug the issue?

---
*Response generated by 🤖 [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*

 ⚠️ This is an AI-generated response. Please verify any critical information.

**Comment by sayinmehmet47** [comment]
This AI generated answer did not solve my problem

**Comment by GreatV** [comment]
You may need to fine-tune your model. please refer to https://paddlepaddle.github.io/PaddleOCR/latest/en/ppocr/model_train/finetune.html

```python
from paddleocr import PaddleOCR
from PIL import Image, ImageDraw, ImageFont

ocr = PaddleOCR(use_angle_cls=True, lang="de", det_db_unclip_ratio=1.5)

img_path = "./Snipaste_2025-03-05_08-48-05.png"
# slice = {'horizontal_stride': 300, 'vertical_stride': 500, 'merge_x_thres': 50, 'merge_y_thres': 35}
# results = ocr.ocr(img_path, cls=True, slice=slice)
results = ocr.ocr(img_path, cls=True)

image = Image.open(img_path).convert("RGB")
draw = ImageDraw.Draw(image)
font = ImageFont.truetype("./doc/fonts/german.ttf", size=10)


for res in results:
    for line in res:
        box = [tuple(point) for point in line[0]]

        box = [(min(point[0] for point in box), min(point[1] for point in box)),
               (max(point[0] for point in box), max(point[1] for point in box))]
        txt = line[1][0]
        draw.rectangle(box, outline="red", width=2)
        draw.text((box[0][0], box[0][1] - 15), txt, fill="blue", font=font) 


image.save("result.jpg")

```

![result](https://github.com/user-attachments/assets/2d494762-01f2-4394-9436-3d3f2b7e01d6)
